test/Sentry.Tests/Protocol/Envelopes/EnvelopeTests.cs

```diff
@@ -492,6 +535,36 @@ public async Task Roundtrip_WithUserFeedback_Success()
             payloadContent.Should().BeEquivalentTo(feedback);
         }
 
+        [Fact]
+        public async Task Roundtrip_WithSession_Success()
+        {
+            // Arrange
+            var sessionUpdate = new Session("foo", "bar", "baz").CreateSnapshot(true);
+
+            using var envelope = Envelope.FromSession(sessionUpdate);
+
+#if !NET461 && !NETCOREAPP2_1
+            await
+#endif
+                using var stream = new MemoryStream();
+
+            // Act
+            await envelope.SerializeAsync(stream);
+            stream.Seek(0, SeekOrigin.Begin);
+
+            using var envelopeRoundtrip = await Envelope.DeserializeAsync(stream);
+
+            // Assert
+
+            // Can't compare the entire object graph because output envelope contains evaluated length,
```

Comment:
Can we remove length and compare?

Reply:
What do you mean? How?

Reply:
What if we do:

Read all bytes from `envelope.GetStream()` it should set `Length`.

The goal is to make sure we are in fact deserializign the thing correctly. This test doesn't verify that.